layersvt/device_simulation.cpp

```diff
@@ -82,12 +83,12 @@ const uint32_t kLayerPropertiesCount = (sizeof(kLayerProperties) / sizeof(kLayer
 const char *kOurLayerName = kLayerProperties[0].layerName;
 
 // Instance extensions that this layer provides:
-const VkExtensionProperties *kInstanceExtensionProperties = {};
-const uint32_t kInstanceExtensionPropertiesCount = (sizeof(kInstanceExtensionProperties) / sizeof(kInstanceExtensionProperties[0]));
+const std::array<VkExtensionProperties,16> kInstanceExtensionProperties = {};
```

Comment:
Concerned about magic numbers (16 here and 15 below) defeating the intent of compile-time auto-sizing.

Reply:
std::vector is likely the best alternative even if it has a heap allocation.
std::array could have the size explicitly set to 0 until someone added an element to it, which would be annoying but then there is no wasted space.

Reply:
An array size of 0 would be an appropriate value, considering it is currently empty.